airflow/contrib/sensors/weekday_sensor.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+from airflow.utils import timezone
+from airflow.utils.decorators import apply_defaults
+from airflow.contrib.utils.weekday import WeekDay
+
+
+class DayOfWeekSensor(BaseSensorOperator):
+    """
+    Waits until the first specified day of the week. For example, if the execution
+    day of the task is '2018-12-22' (Saturday) and you pass 'FRIDAY', the task will wait
+    until next Friday.
+
+    :param week_day: Day of the week (full name). Example: "MONDAY"
+    :type week_day_number: str
+    :param use_task_execution_day: If ``True``, uses task's execution day to compare
+        with week_day_number. Execution Date is Useful for backfilling.
+        If ``False``, uses system's day of the week. Useful when you
+        don't want to run anything on weekdays on the system.
+    :type use_task_execution_day: bool
+    """
+
+    @apply_defaults
+    def __init__(self, week_day,
+                 use_task_execution_day=False,
+                 *args, **kwargs):
+        super(DayOfWeekSensor, self).__init__(*args, **kwargs)
+        self.week_day = week_day
+        self.use_task_execution_day = use_task_execution_day
```

Comment:
Shouldn't the sensor use the execution date by default? (As Airflow's model is built around execution dates, I think that they should be preferred over `datetime.utcnow`. In that case, I would change the argument to `now`, with a value of `True` indicating that the now time should be used (and the default being False).

Reply:
The idea behind that having current date as default (`use_task_execution_day`=`False`) - most of the use-cases for any of these sensors are to stop until a certain day. For e.g. for weekdaysensor - the users want to run the DAG on weekends when everyone is on holiday. 

Also, all of our current sensors use current date (as compared to execution_date) - but like you I wanted an option to use execution_date as well as Airflow model is certainly built around execution dates, hence I added this functionality to at least this 2 sensors for now.